apps/mark/frontend/src/app_quit_on_idle.test.tsx

```diff
@@ -59,20 +59,19 @@ test('Insert Card screen idle timeout to quit app', async () => {
     />
   );
 
-  await advanceTimersAndPromises();
-
   // Ensure we're on the Insert Card screen
-  screen.getByText('Insert Card');
+  await screen.findByText('Insert Card');
 
   expect(window.kiosk?.quit).not.toHaveBeenCalled();
 
   // Check that we requested a quit after the idle timer fired.
-  await advanceTimersAndPromises();
   await advanceTimersAndPromises(QUIT_KIOSK_IDLE_SECONDS);
-  expect(window.kiosk?.quit).toHaveBeenCalledTimes(1);
+  await waitFor(() => {
+    expect(window.kiosk?.quit).toHaveBeenCalledTimes(1);
+  });
 });
 
-test('Voter idle timeout', async () => {
+test.skip('Voter idle timeout', async () => {
```

Comment:
This is the thorniest test failure of the them all. It appears that once the idle timeout passes, this `useEffect` in the shared `IdlePage` component runs many, many times: https://github.com/votingworks/vxsuite/blob/3ee24946c1b0271928b6787dabf327a45fc1f237/libs/mark-flow-ui/src/pages/idle_page.tsx#L37-L42
Setting `isLoading` to `true` _should_ prevent the effect from re-running, but for some reason that state change doesn't propagate fast enough.

I tried this out in the app, in `kiosk-browser`, and found that it worked as expected. But I also noticed that the tree was re-rendering a ton, and traced it to this callback: https://github.com/votingworks/vxsuite/blob/3ee24946c1b0271928b6787dabf327a45fc1f237/apps/mark/frontend/src/app_root.tsx#L520-L526
This is similar to we discussed here: https://github.com/votingworks/vxsuite/pull/3191#issuecomment-1502308643. Mutations returned from `useMutation` actually change on every render. And for some reason the app is re-rendering a lot. What I think is happening (due to React 18 concurrency and batching) is that the mutation within the `IdlePage` `useEffect` is triggering a circular issue where the mutation changes, the dependencies change, and the `useEffect` runs again before batched state changes take effect, so `isLoading` is still false.

One solution may be to have a ref for `isLoading`. In a subsequent PR, I fix by removing the mutation as a dependency of the callback, instead using the `mutate` function of the mutation.

Thoughts?

Reply:
We could move the logic of that effect into the `useInterval`.

I also think we need to remove the mutation object from the dependency array like you mentioned. We should never be doing that, it's a footgun.

Reply:
Excellent suggestion! That works. Fixed and pushed.

I created a ticket for the fix / lint rule for `Mutation` dependencies: https://github.com/votingworks/vxsuite/issues/3794